Source/Core/Core/PowerPC/Jit64/Jit_Integer.cpp

```diff
@@ -1735,32 +1735,47 @@ void Jit64::rlwimix(UGeckoInstruction inst)
       {
         MOV(32, R(RSCRATCH), Rs);
         SHL(32, R(RSCRATCH), Imm8(inst.SH));
-        AndWithMask(Ra, ~mask);
-        OR(32, Ra, R(RSCRATCH));
       }
       else if (right_shift)
       {
         MOV(32, R(RSCRATCH), Rs);
         SHR(32, R(RSCRATCH), Imm8(32 - inst.SH));
-        AndWithMask(Ra, ~mask);
-        OR(32, Ra, R(RSCRATCH));
       }
       else
       {
         RotateLeft(32, RSCRATCH, Rs, inst.SH);
-        XOR(32, R(RSCRATCH), Ra);
-        AndWithMask(RSCRATCH, mask);
-        XOR(32, Ra, R(RSCRATCH));
+      }
+
+      if (mask == 0xFF || mask == 0xFFFF)
+      {
+        MOV(mask == 0xFF ? 8 : 16, Ra, R(RSCRATCH));
```

Comment:
Does this really work? I would have expected MOVZX.

Reply:
Intended behaviour is `Ra &= 0xFFFF0000; Ra |= RSCRATCH & 0xFFFF`, movzx is not appropriate here.